src/java/za/co/mip/ablduck/utilities/HTMLGenerator.java

```diff
@@ -0,0 +1,343 @@
+/**
+ * Copyright 2017 MIP Holdings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+package za.co.mip.ablduck.utilities;
+
+import java.util.List;
+import java.util.Map;
+import java.util.ResourceBundle;
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+
+import org.apache.tools.ant.BuildException;
+
+import com.google.gson.annotations.SerializedName;
+
+import za.co.mip.ablduck.ABLDuck;
+import za.co.mip.ablduck.models.SourceJSObject;
+import za.co.mip.ablduck.models.generic.DeprecatedObject;
+import za.co.mip.ablduck.models.generic.MetaObject;
+import za.co.mip.ablduck.models.source.MemberObject;
+import za.co.mip.ablduck.models.source.ParameterObject;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.text.MessageFormat;
+
+public class HTMLGenerator {
+    private static final ResourceBundle TEMPLATES = ResourceBundle
+            .getBundle("za.co.mip.ablduck.utilities.templates");
+
+    private static Pattern returnType = Pattern.compile("\\):(.*)");
+    private Map<String, SourceJSObject> classes;
+
+    public String getTemplate(String key) {
+        return TEMPLATES.getString(key);
+    }
+
+    public String getClassHtml(Map<String, SourceJSObject> allclasses, SourceJSObject cls) {
+        this.classes = allclasses;
+
+        return MessageFormat.format(getTemplate("CLASSBODY"), renderSidebar(cls),
+                renderClassComment(cls), renderMemberDetails(cls, "property", "Properties"),
+                renderMemberDetails(cls, "method", "Methods"));
+    }
+
+    private String renderSidebar(SourceJSObject cls) {
+        boolean hasSubIntImp = !cls.subclasses.isEmpty() || !cls.interfaces.isEmpty()
+                || !cls.implementers.isEmpty();
+        if (hasSubIntImp || !"".equals(cls.ext) || !"".equals(cls.author))
+            return MessageFormat.format(getTemplate("SIDEBAR"), renderClassTree(cls),
+                    renderSubclasses(cls), renderInterfaces(cls), renderImplementers(cls),
+                    renderAuthor(cls));
+
+        return "";
+    }
+
+    private String renderImplementers(SourceJSObject cls) {
+        if (!cls.implementers.isEmpty()) {
+            StringBuilder implementerBuilder = new StringBuilder();
+            for (String implementer : cls.implementers) {
+                implementerBuilder.append(MessageFormat.format(getTemplate("DEPENDANCY.ITEM"),

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "DEPENDANCY.ITEM" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1192)